src/Event.php

```diff
@@ -24,10 +25,17 @@ final class Event implements \JsonSerializable
     private $id;
 
     /**
-     * @var string The date and time of when this event was generated
+     * @var string|float The date and time of when this event was generated
      */
     private $timestamp;
 
+    /**
+     * This property is used if it's a Transaction event together with $timestamp it's the duration of the transaction.
+     *
+     * @var string|float|null The date and time of when this event was generated
+     */
+    private $startTimestamp;
```

Comment:
Instead of having such a polymorphic behavior, maybe we can split it in two class, with a possible common ancestor? WDYT?